Fix inventory cache interface (#50446)
ansible/ansible: docs/docsite/rst/porting_guides/porting_guide_2.8.rst

     removed in 2.12.  If you need the old behaviour switch to ``FactCache.first_order_merge()``
     instead.

> * Inventory plugins writers should know that supporting file-backed caching via self.cache is deprecated and will

Reviewer: ```suggestion
* Supporting file-backed caching via self.cache is deprecated and will
```

>   be removed in Ansible 2.12. Instead, self._cache should be used as a dictionary. For implementation details, see

Reviewer: ```suggestion
  be removed in Ansible 2.12. If you maintain an inventory plugin, update it to use ``self._cache`` as a dictionary. For implementation details, see
```

>   the developer guide on inventory plugins.

Reviewer: ```suggestion
  the :ref:`developer guide on inventory plugins<inventory_plugin_caching>`.
```

> * Importing cache plugins directly is deprecated and will be removed in Ansible 2.12. The plugin_loader should be

Reviewer: ```suggestion
* Importing cache plugins directly is deprecated and will be removed in Ansible 2.12. Use the plugin_loader
```

>   used so direct options, environment variables, and other means of configuration can be reconciled using the config

Reviewer: ```suggestion
  so direct options, environment variables, and other means of configuration can be reconciled using the config
```

* Play recap now counts ``ignored`` and ``rescued`` tasks as well as ``ok``, ``changed``, ``unreachable``, ``failed`` and ``skipped`` tasks, thanks to two additional stat counters in the ``default`` callback plugin. Tasks that fail and have ``ignore_errors: yes`` set are listed as ``ignored``. Tasks that fail and then execute a rescue section are listed as ``rescued``. Note that ``rescued`` tasks are no longer counted as ``failed`` as in Ansible 2.7 (and earlier).

> * Cache plugins can now be used for caching inventory. To make a new or custom cache plugin compatible with inventory

Reviewer: This section seems awfully long for the porting guide. I'd move anything that isn't already covered in the `developing_inventory` page to there, and link to it from here. That way folks will be able to find this info easily even when 2.8 starts to fade in the rear-view mirror.

Author: Good point!
